src/panels/lovelace/cards/hui-fan-card.ts

```diff
@@ -0,0 +1,357 @@
+import { FanCardConfig } from "./types";
+
+@customElement("hui-fan-card")
+export class HuiFanCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import(
+      /* webpackChunkName: "hui-fan-card-editor" */ "../editor/config-elements/hui-fan-card-editor"
+    );
+    return document.createElement("hui-fan-card-editor");
+  }
+
+  public static getStubConfig(
+    hass: HomeAssistant,
+    entities: string[],
+    entitiesFallback: string[]
+  ): FanCardConfig {
+    const includeDomains = ["fan"];
+    const maxEntities = 1;
+    const foundEntities = findEntities(
+      hass,
+      maxEntities,
+      entities,
+      entitiesFallback,
+      includeDomains
+    );
+
+    return { type: "fan", entity: foundEntities[0] || "" };
+  }
+
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @internalProperty() private _config?: FanCardConfig;
+
+  private _speedTimeout?: number;
+
+  public getCardSize(): number {
+    return 5;
+  }
+
+  public setConfig(config: FanCardConfig): void {
+    if (!config.entity || config.entity.split(".")[0] !== "fan") {
+      throw new Error("Specify an entity from within the fan domain.");
+    }
+
+    this._config = {
+      tap_action: { action: "toggle" },
+      hold_action: { action: "more-info" },
+      ...config,
+    };
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config!.entity] as FanEntity;
+
+    if (!stateObj) {
+      return html`
+        <hui-warning>
+          ${createEntityNotFoundWarning(this.hass, this._config.entity)}
+        </hui-warning>
+      `;
+    }
+
+    const max_speed = stateObj.attributes.speed_list.length;
+
+    const speed = stateObj.attributes.speed_list.indexOf(
+      stateObj.attributes.speed
+    );
+
+    return html`
+      <ha-card
+        .header=${this._config!.title}
+        class=${classMap({ "no-header": !this._config!.title })}
+      >
+        <mwc-icon-button
+          class="more-info"
+          label="Open more info"
+          @click=${this._handleMoreInfo}
+          tabindex="0"
+        >
+          <ha-svg-icon .path=${mdiDotsVertical}></ha-svg-icon>
+        </mwc-icon-button>
+
+        <div class="content">
+          <div id="controls">
+            <div id="slider">
+              <round-slider
+                min="0"
+                max=${max_speed}
+                .value=${speed}
+                .disabled=${UNAVAILABLE_STATES.includes(stateObj.state)}
+                @value-changing=${this._dragEvent}
+                @value-changed=${this._setSpeed}
+                style=${styleMap({
+                  visibility: "visible",
+                })}
+              ></round-slider>
+              <ha-icon-button
+                class="fan-button ${classMap({
+                  "slider-center": true,
```

Comment:
no need for a classmap for a class that is always true. Just add it like you have `fan-button`